engine/robot_logic/traversal.py

```diff
@@ -64,17 +68,34 @@ def move_to_target_node(robot_state, target, allowed_dist_error, database):
     robot_state.goal_location = target
     robot_state.loc_pid_x.reset_integral()
     robot_state.loc_pid_y.reset_integral()
+    using_heading_pid = True
+    simulate = False
+    if simulate:
+        i = 0
+    import matplotlib.pyplot as plt
     while (distance_away > allowed_dist_error) and not (robot_state.phase == Phase.AVOID_OBSTACLE):
         # Error in terms of latitude and longitude, NOT meters
         x_coords_error = target[0] - robot_state.state[0]
         y_coords_error = target[1] - robot_state.state[1]
+        if simulate:
+            i += 1
+            if i == 50:
```

Comment:
why is i = 50 resulting in the break?

Reply:
we're limiting the number of points we are plotting. If we don't limit it, then the graphing would take forever because there is a lot of iterations